### PR TITLE
ST-2195: Refactor NewLoginService() to a reusable library

### PR DESCRIPTION
**Pull request by yarencheng**
Main changes:

* Export `*go-restful.Webservice` by `login.New(...).WebService()`
  * aurora and vortex projects are using `*go-restful.Webservice`. Keep to use it in the futrue.
* Reorganize files
  * Copy related file from oauth repo. E.g. entity/user.go
  * move `webservice/acl/*` to `webservice/login/.`
* Use a `UserStorage interface{}` to separate `http handler` and `mongo session`
  * use `UserStorage interface{}` in `webservice/login/session/*.go`
  * Remove useless or duplicated code in `webservice/login/session/*.go`
   * For local testing, a in-memory version `webserviceuserstorage/memory.go`
* Fix `panic()` caused by nil pointer

### CODE REVIEW

File: login/config/config.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package config
+
+type Type string
+
+const (
+	MEMORY Type = "MEMORY"
```
Comment (c9s) [inline login/config/config.go]: Don't use full upper case for constant. Use `Memory`, `Mongo` ... etc

File: login/entity/notification_preference.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// and determine what kind of messages will be sent.
+type NotificationPreference struct {
+	EnableSMS   bool `json:"enableSMS" bson:"enableSMS"`
+	EnableEmail bool `json:"enableEmail" bson:"enableEmail"`
+
+	WhenJobStart   bool `json:"whenJobStart" bson:"whenJobStart"`
+	WhenJobSuccess bool `json:"whenJobSuccess" bson:"whenJobSuccess"`
+	WhenJobFail    bool `json:"whenJobFail" bson:"whenJobFail"`
+	WhenJobStop    bool `json:"whenJobStop" bson:"whenJobStop"`
+	WhenJobDelete  bool `json:"whenJobDelete" bson:"whenJobDelete"`
```
Comment (c9s) [inline login/entity/notification_preference.go]: This should be not here.

File: login/entity/session_response.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package entity
+
+type SessionResponse struct {
+	ID          string `json:"id,omitempty"`
+	Token       string `json:"token,omitempty"`
+	ExpiredAt   int64  `json:"expiredAt,omitempty"`
+	CurrentUser User   `json:"currentUser,omitempty"`
+}
```
Comment (c9s) [inline login/entity/session_response.go]: Things are not stored in the db should not be in the entity package.

File: login/entity/signIn_response.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package entity
+
+type SignInResponse struct {
+	Error        bool            `json:"error"`
+	AuthRequired bool            `json:"authenRequired,omitempty"`
+	Message      string          `json:"message"`
+	SignInUrl    string          `json:"signInUrl,omitempty"`
+	Session      SessionResponse `json:"session,omitempty"`
+}
```
Comment (c9s) [inline login/entity/signIn_response.go]: Things are not stored in the db should not be in the entity package.

File: login/entity/user.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	VerificationCode      string        `bson:"verification_code" json:"verification_code"`
+	Jwt                   string        `bson:"jwt" json:"jwt"`
+	AccessToken           string        `bson:"access_token" json:"access_token"`
+	AccessTokenExpiryTime int64         `bson:"access_token_expiry_time" json:"access_token_expiry_time"`
+	RefreshToken          string        `bson:"refresh_token" json:"refresh_token"`
+	CreatedAt             int64         `bson:"created_at,omitempty" json:"created_at"`
+	UpdatedAt             int64         `bson:"updated_at,omitempty" json:"updated_at"`
+	LastLoggedInAt        int64         `bson:"last_loggedin_at,omitempty" json:"last_loggedin_at"`
+	Revoked               bool          `bson:"revoked" json:"revoked"`
+	JobPriority           float64       `bson:"job_priority" json:"job_priority"`
```
Comment (c9s) [inline login/entity/user.go]: This is not related. should be removed

**Review by c9s — CHANGES_REQUESTED** [review]

**Review by c9s — APPROVED** [review]

**Review by hwchiu — APPROVED** [review]
LGTM

**Pull request merged**